src/IECoreDelight/Renderer.cpp

```diff
@@ -1028,6 +1046,176 @@ class DelightObject : public IECoreScenePreview::Renderer::ObjectInterface
+			: m_transformHandle( context, name, ownership, "transform", {} ), m_instanceCache( instanceCache ), m_usdLightType( USDLightType::None), m_ownership( ownership ), m_haveTransform( false )
+		{
+			NSIConnect(
+				m_transformHandle.context(),
+				m_transformHandle.name(), "",
+				NSI_SCENE_ROOT, "objects",
+				0, nullptr
+			);
+		}
+
+		void transform( const Imath::M44f &transform ) override
+		{
+			if( transform == M44f() && !m_haveTransform )
+			{
+				return;
+			}
+
+			M44d m( transform );
+			NSIParam_t param = {
+				"transformationmatrix",
+				m.getValue(),
+				NSITypeDoubleMatrix,
+				0, 1, // array length, count
+				0 // flags
+			};
+			NSISetAttribute( m_transformHandle.context(), m_transformHandle.name(), 1, &param );
+
+			m_haveTransform = true;
+		}
+
+		void transform( const std::vector<Imath::M44f> &samples, const std::vector<float> &times ) override
+		{
+			if( m_haveTransform )
+			{
+				NSIDeleteAttribute( m_transformHandle.context(), m_transformHandle.name(), "transformationmatrix" );
+			}
+
+			for( size_t i = 0, e = samples.size(); i < e; ++i )
+			{
+				M44d m( samples[i] );
+				NSIParam_t param = {
+					"transformationmatrix",
+					m.getValue(),
+					NSITypeDoubleMatrix,
+					0, 1, // array length, count
+					0 // flags
+				};
+				NSISetAttributeAtTime( m_transformHandle.context(), m_transformHandle.name(), times[i], i, &param );
+			}
+
+			m_haveTransform = true;
+		}
+
+		bool attributes( const IECoreScenePreview::Renderer::AttributesInterface *attributes ) override
+		{
+			if( m_attributes )
+			{
+				if( attributes == m_attributes )
+				{
+					return true;
+				}
+
+				NSIDisconnect(
+					m_transformHandle.context(),
+					m_attributes->handle().name(), "",
+					m_transformHandle.name(), "geometryattributes"
+				);
+			}
+
+			m_attributes = static_cast<const DelightAttributes *>( attributes );
+			NSIConnect(
+				m_transformHandle.context(),
+				m_attributes->handle().name(), "",
+				m_transformHandle.name(), "geometryattributes",
+				0, nullptr
+			);
+
+			DelightHandleSharedPtr newInstance;
+
+			if( const ShaderNetwork *usdLightShader = m_attributes->usdLightShader() )
+			{
+				const Shader *light = usdLightShader->outputShader();
+				const USDLightType prevUSDLightType = m_usdLightType;
+
+				/// \todo Would all of this be better housed in a `LightAlgo.cpp` or similar?
+				if( light->getName() == "SphereLight" )
+				{
+					m_usdLightType = USDLightType::SphereLight;
+
+					const auto radiusData = light->parametersData()->member<FloatData>( g_radiusParameterName );
+					const float radius = radiusData->readable();
+
+					const auto prevRadiusData = m_geometryParameters.member<FloatData>( g_radiusParameterName );
+					if( !prevRadiusData || prevRadiusData->readable() != radius || prevUSDLightType != m_usdLightType )
+					{
+						IECore::ConstObjectPtr geo = new SpherePrimitive( radiusData->readable() );
+						newInstance = m_instanceCache->get( geo.get() );
```

Comment:
Since we're already doing the "is it the right geometry type?" juggling, and since light geometry isn't heavy, I'd be tempted to forgo the InstanceCache and just make the geometry directly via NSI, and then update the radius directly when necessary, without throwing away and recreating the geometry.

Reply:
That sounds good, and especially now that the geometry logic is one step deeper below `DelightLight`, removing the need for holding on to the instancer cache removes the need to pass it along down the stack.